src/main/webapp/app/components/tabs/CurationToolsTab.tsx

```diff
@@ -1,105 +1,142 @@
 import { IRootStore } from 'app/stores';
 import React, { useEffect, useRef, useState } from 'react';
-import { FaExclamationCircle } from 'react-icons/fa';
-import { Button, Col, Row } from 'reactstrap';
-import './curation-tools-tab.scss';
+import { Button, Row } from 'reactstrap';
+import { FaRegCheckCircle } from 'react-icons/fa';
+import { FaRegCircleXmark } from 'react-icons/fa6';
 import { notifyError } from 'app/oncokb-commons/components/util/NotificationUtils';
-import { isPromiseOk } from 'app/shared/util/utils';
-import { getErrorMessage } from 'app/oncokb-commons/components/alert/ErrorAlertUtils';
-import { observer } from 'mobx-react';
+import { IGene } from 'app/shared/model/gene.model';
+import _ from 'lodash';
+import { IFlag } from 'app/shared/model/flag.model';
 
-function CurationToolsTab({ addMetaListener, metaData, createMetaGene, deleteMetaGene, createGene, deleteGene }: StoreProps) {
-  const selectedGene = useRef<string>(null);
-  const [createButtonDisabled, setCreateButtonDisabled] = useState(true);
-  const [showGeneExistsWarning, setShowGeneExistsWarning] = useState(false);
+export function CurationToolsTab({ metaList, addMetaListListener, gene, geneEntities, searchGenes, updateGene, searchFlags }: StoreProps) {
+  const reviewList = metaList?.[gene.name].review;
+  const needsReview = reviewList ? Object.keys(reviewList).some(key => key !== 'currentReviewer' && reviewList[key]) : false;
+
+  const geneSummaryEmpty = !gene.summary;
+
+  const geneBackgroundEmpty = !gene.background;
+
+  const geneTypeEmpty = !gene.type.ocg && !gene.type.tsg;
+
+  const confirmButtonDisabled = needsReview || geneSummaryEmpty || geneBackgroundEmpty || geneTypeEmpty;
+
+  const [isReleased, setIsReleased] = useState<boolean>(null);
+  const [releaseGeneClicked, setReleaseGeneClicked] = useState(false);
+
+  const geneToUpdate = useRef<IGene>(null);
 
   useEffect(() => {
-    const callback = addMetaListener();
+    const callback = addMetaListListener();
 
-    return () => callback && callback();
+    return () => {
+      callback && callback();
+    };
   }, []);
 
-  function handleChangeSelectedGene(option) {
-    const gene = option?.label;
-    selectedGene.current = gene;
-
-    if (!gene) {
-      setCreateButtonDisabled(true);
-      setShowGeneExistsWarning(false);
-    } else if (Object.keys(metaData).includes(gene)) {
-      setCreateButtonDisabled(true);
-      setShowGeneExistsWarning(true);
-    } else {
-      setCreateButtonDisabled(false);
-      setShowGeneExistsWarning(false);
-    }
+  function isReleasedFlag(flag: IFlag) {
+    return flag.type === 'GENE_PANEL' && flag.flag === 'ONCOKB';
+  }
+
+  useEffect(() => {
+    const geneData = geneEntities.find(entity => entity.hugoSymbol === gene.name);
+    setIsReleased(geneData?.flags?.some(flag => isReleasedFlag(flag)) || false);
+    geneToUpdate.current = geneData;
+  }, []);
+
+  function getStatusIcon(checkPassed: boolean) {
```

Comment:
Given all use cases within this file, let's rename the pass to checkNotPassed and update the usage.

Reply:
After defining the tests to be run when creating a gene, we now have a `passed` variable for each test. The use cases all now use `getStatusIcon(test.passed)`, so will leave the same